### PR TITLE
Confirm message for ctrl+c

### LINKED ISSUE #762: Confirm exit on ctrl + c
If a user accidentally press `ctrl+c`, the server immediately stops. It would be better to have a message to confirm the exit. <br><br>`Shutdown libertem server (y/[n])?`

### PR DESCRIPTION
**Pull request by twentyse7en**
Ask for confirmation if the user press ctrl+c.
can immediately stop using another ctrl+c.
willn't work on windows.

fixes #762 
## Contributor Checklist:

* [x] I have added or updated my entry in [the creators.json file](https://github.com/LiberTEM/LiberTEM/blob/master/packaging/creators.json)
* [ ] I have added [a changelog entry](https://github.com/LiberTEM/LiberTEM/tree/master/docs/source/changelog) for my contribution
* [ ] I have added/updated documentation for all user-facing changes

### CODE REVIEW

**Comment by twentyse7en** [comment]
I referred Jupyter for implementing this. Windows situation is handled separately as `select.select()` willn't work on windows.

File: src/libertem/web/server.py [hunk 1]
```diff
@@ -130,6 +133,43 @@ def main(host, port, numeric_level, event_registry, shared_state):
...
+    loop = tornado.ioloop.IOLoop.current()
+    # register more forceful signal handler for ^C^C case
+    signal.signal(signal.SIGINT, partial(sig_exit, shared_state=shared_state))
+    thread = threading.Thread(target=partial(_confirm_exit, shared_state, loop))
+    thread.daemon = True
+    thread.start()
+
+
+def handle_signal(shared_state):
+    if not sys.platform.startswith('win'):
```
Comment (twentyse7en) [inline src/libertem/web/server.py:167]: [jupyter](https://github.com/jupyter/notebook/blob/e577b6c40c194d70c058f30aee33ba4fca5502b9/notebook/notebookapp.py#L1578) also check for `sys.stdin` and `sys.stdin.isatty()` , do we need this ?

Comment (sk1p) [reply src/libertem/web/server.py:167]: Yes: if the server is running in the background, detached from `stdin`, we can't ask the user for confirmation.

I'll do a more thorough review beginning next week

**Comment by sk1p** [comment]
> I referred Jupyter for implementing this.

One thing to be careful of, when referencing or even copying code, is that licensing is taken care of. In this case, have a look at [the LICENSE file of jupyter notebook](https://github.com/jupyter/notebook/blob/master/LICENSE) and see what is to be done. In case of taking significant amounts of code or ideas from code, we should put a license reference into the documentation, [like I did for the SEQ file format support](https://libertem.github.io/LiberTEM/acknowledgments.html#code-licensing). This also always depends on the concrete licenses.

> Windows situation is handled separately as `select.select()` willn't work on windows.

Okay :+1:

File: src/libertem/web/server.py [hunk 2]
```diff
@@ -130,6 +133,43 @@ def main(host, port, numeric_level, event_registry, shared_state):
     return app
 
 
+def _confirm_exit(shared_state, loop):
+    log.info('interrupted, ')
```
Comment (sk1p) [inline src/libertem/web/server.py:137]: I think the `, ` at the end of the log message can be removed.

**Comment by sk1p** [comment]
The flake8 error is not your fault, that's because of a new release of flake8 that supports more checks. If you like, you can fix it, otherwise I'll merge. Thanks!

**Comment by sk1p** [comment]
Thanks, merged!

**Pull request merged**